app/controllers/passwords_controller.rb

```diff
@@ -0,0 +1,49 @@
+class PasswordsController < ApplicationController
+  def forgot_password
+  end
+
+  def request_token
+    @user = User.find_by(email_address: params[:email])
+    @token = SecureRandom.urlsafe_base64
```

Comment:
This is not a big deal, but I would probably put the token generation after the user validation. There is no need to generate a token if you are not going to use it. When an app goes large scale, every saved cpu cycle helps the app run smoothly!